transcom/mymove
# B-21337-INT bulk assignment data fetch setup

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21337)

## Summary

LOTS GOING ON HERE FRIENDS. This is the initial setup for the endpoint to get bulk assignment data for the bulk assignment modal, and implements getting the data for the COUNSELING queue.

Normal endpoint stuff: Part of the queues handler. Added a switch in because this endpoint will handle retrieving data for all the queues. Fetch the users with their workload (amount of moves they are currently working, so the super visor bulk assigning moves can better divy the work out), and then fetch the moves. Returning IDs because thats what the gov wants (they would like to lock down the moves while the super visor is bulk assigning, which will come in 06032.

## TO TEST
1. Pick a transportation office that provides services counseling, and make some SCs there. Make a super visor there. Name the supervisor after yourself so you can feel like a boss.
2. Create some moves or update some moves in the database to have their counseling office = your transportation office.
3. Assign some of your underlings to moves, keeping track of how many moves are counseled by your transportation office, how many are assigned, how many are unassigned, how many moves each of your underlings are working.
4. View the bulk assignment data and confirm that the data returned is accurate. What we are looking for is:
-an array of `BulkAssignmentMoveIDs`, ordered by the earliest move date
-an array of `availableOfficeUsers`, with office user's name, ID, and workload (the current amount of moves they are assigned to)

If you're lazy like me just throw this into ServicesCounselingQueue.jsx and you should be good
`const { bulkAssignmentData } = useBulkAssignmentQueries('COUNSELING');`
